iib/web/models.py

```diff
@@ -432,20 +462,35 @@ def add_architecture(self, arch_name):
         if arch not in self.architectures:
             self.architectures.append(arch)
 
-    @classmethod
-    def from_json(cls, kwargs):
+    @abstractmethod
+    def from_json(
+        cls,
+        kwargs: PayloadTypesUnion,
+    ) -> 'Request':
```

Comment:
As you've set `from __future__ import annotations` you can change this to:

```suggestion
    ) -> Request:
```

Reply:
Thanks @JAVGan , I pushed the commit, please check the changes, I will squash the commits then.

Reply:
@xDaile LTGM, you can squash it 👍🏻